Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1143,8 +1143,10 @@ class BrowserViewController: UIViewController {
     func finishEditingAndSubmit(_ url: URL, visitType: VisitType, forTab tab: Tab) {
         urlBar.currentURL = url
         leaveOverlayMode(didCancel: false)
-
-        if let nav = tab.loadRequest(URLRequest(url: url)) {
+        // Ecosia: enriching the app with a language region header for market selection options
+        var urlRequest = URLRequest(url: url)
+        urlRequest.addLanguageRegionHeader()
```

Comment:
To give you some more context. We don't need to add the region header for all requests, but only for the request that reach our ecosia domains. Thus it's not the right place to add them here. Instead I'd suggest to use the code parts where also our `ecosify()` method is called.

Reply:
TIL💡! Thanks! I thought that these headers would have been added for both scenarios:
- typing something into the URL bar
- clicking on a suggestion out of the typed text in the URL Bar